client.go

```diff
@@ -468,17 +469,10 @@ func (c *Client) runPublisher(ouputChan *amqp.Channel) {
 				return
 			}
 
-			if !c.confirmMode {
-				// We're not in confirm mode so we confirm that we have sent
-				// the request here.
-				c.confirmRequest(request)
```

Comment:
This is weird, trying to understand why we did this. ~~If we tolt the server to confirm requests, will it just hang on `PublishWithContext` til it's confirmed? Because the current code doesn't wait for or read anything, we just always publish to this channel which seems useless?~~ I see now we run a confirms worker which didn't change my opinion of this being odd and not adding value 😄 

~~Anyways, since the condition now checks for _either_, does this mean that if `c.confirmMode` is true but `reqeust.Reply` is false, we won't respond here and thus hang? Until when? Reading L738, it looks like we'll wait til someone publishes on `r.response` and if we don't do it here, it looks like we will wait til we get a response back and publish on L611? It doesn't sound like the expected behavior if `request.Reply` is false but we want to use `confirmMode`?~~

Reply:
Ah, I see now that we already had code for this on (now) L602! That makes my original question even weirder, if we _did_ wait for the confirm and we did reply to it with `nil` if we didn't want reply, why the heck did we have a separate confirms channel that we didn't need to use?

Oh well, that's history, great change!

Reply:
Yes, we always waited for confirm before waiting for the reply. So here, we "fake" it by confirming immediately if we where not running in confirm mode. Anyway, that is history now since we no longer have a separate confirms channel.